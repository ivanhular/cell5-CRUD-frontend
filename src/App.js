import React, { useState, useEffect, useContext } from 'react'
import { Container, Row } from 'react-bootstrap'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlus } from '@fortawesome/free-solid-svg-icons'
import HobbyList from './components/HobbyList'
import HobbyModal from './components/HobbyModal'
import HobbySearchBar from './components/HobbySearchBar'
import Alert from './components/Alert'
import HobbyDeleteModal from './components/HobbyDeleteModal'
import HobbyContentModal from './components/HobbyContentModal'
import { GlobalProvider } from './context/Provider'
import './App.css'

const styles = {
  app: {
    marginTop: '5em',
  },
  container: {
    maxWidth: '900px',
  },
}

const App = () => {
  return (
    <GlobalProvider>
      <div className='App' style={styles.app}>
        <Container style={styles.container}>
          <Row className='pb-3'>
            <HobbySearchBar />
          </Row>
          <Row>
            <HobbyList />
          </Row>
        </Container>
      </div>
      <Alert />
      <HobbyModal />
      <HobbyDeleteModal />
      <HobbyContentModal />
    </GlobalProvider>
  )
}

export default App
