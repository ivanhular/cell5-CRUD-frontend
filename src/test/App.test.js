import React from 'react'
import { shallow, mount } from '../enzyme'
import App from '../App'

let wrapper

beforeEach(() => {
  wrapper = shallow(<App />)
})

test('should render App correctly', () => {
  expect(wrapper).toMatchSnapshot()
})
