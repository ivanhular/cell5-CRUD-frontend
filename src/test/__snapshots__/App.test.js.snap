// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly 1`] = `
<GlobalProvider>
  <div
    className="App"
    style={
      Object {
        "marginTop": "5em",
      }
    }
  >
    <Container
      fluid={false}
      style={
        Object {
          "maxWidth": "900px",
        }
      }
    >
      <Row
        className="pb-3"
        noGutters={false}
      >
        <HobbySearchBar />
      </Row>
      <Row
        noGutters={false}
      >
        <HobbyList />
      </Row>
    </Container>
  </div>
  <Alert />
  <HobbyModal />
  <HobbyDeleteModal />
  <HobbyContentModal />
</GlobalProvider>
`;
